extensions/data-workspace/src/dataworkspace.d.ts

```diff
@@ -115,4 +120,27 @@ declare module 'dataworkspace' {
 		 */
 		element: any;
 	}
+
+	export interface IProjectAction {
+		/**
+		 * id and display name of the project action
+		 */
+		readonly id: string;
+
+		/**
+		 * icon path of the project action
+		 */
+		readonly icon?: string | vscode.Uri | { light: string | vscode.Uri, dark: string | vscode.Uri };
+
+		/**
+		 * boolean to specify if a toolbar separator should be added after this action
+		 */
+		readonly toolbarSeparatorAfter?: boolean;
```

Comment:
this is kind of hacky :), you need to introduce a group concept.
```
export interface IProjectActionGroup {
    actions: IProjectAction[];
}
```

projectActions's type should be `(IProjectAction | IProjectActionGroup)[]`

Reply:
Done.